src/adapter/etl-adapter-xml/src/Flow/ETL/Adapter/XML/XMLLoader.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\XML;
+
+use function VeeWee\Xml\Writer\Builder\children;
+use function VeeWee\Xml\Writer\Builder\element;
+use function VeeWee\Xml\Writer\Builder\value;
+use function VeeWee\Xml\Writer\Mapper\memory_output;
+use Flow\ETL\Exception\RuntimeException;
+use Flow\ETL\Filesystem\Path;
+use Flow\ETL\Filesystem\Stream\FileStream;
+use Flow\ETL\FlowContext;
+use Flow\ETL\Loader;
+use Flow\ETL\Loader\Closure;
+use Flow\ETL\Partition;
+use Flow\ETL\Rows;
+use VeeWee\Xml\Writer\Writer;
+
+final class XMLLoader implements Closure, Loader, Loader\FileLoader
+{
+    /**
+     * @var array<string, int>
+     */
+    private array $writes = [];
+
+    public function __construct(private readonly Path $path)
+    {
+        if ($this->path->isPattern()) {
+            throw new \InvalidArgumentException("XMLLoader path can't be pattern, given: " . $this->path->path());
+        }
+    }
+
+    public function closure(FlowContext $context) : void
+    {
+        foreach ($context->streams() as $stream) {
+            if ($stream->path()->extension() === 'xml') {
+                \fwrite($stream->resource(), '</rows>');
+            }
+        }
+
+        $context->streams()->close($this->path);
+    }
+
+    public function destination() : Path
+    {
+        return $this->path;
+    }
+
+    public function load(Rows $rows, FlowContext $context) : void
+    {
+        $this->write($rows, $rows->partitions()->toArray(), $context);
+    }
+
+    /**
+     * @param array<Partition> $partitions
+     */
+    private function write(Rows $rows, array $partitions, FlowContext $context) : void
+    {
+        $streams = $context->streams();
+
+        if (!$streams->isOpen($this->path, $partitions)) {
+            $stream = $streams->open($this->path, 'xml', $context->appendSafe(), $partitions);
+
+            if (!\array_key_exists($stream->path()->uri(), $this->writes)) {
+                $this->writes[$stream->path()->uri()] = 0;
+            }
+
+            \fwrite($stream->resource(), '<rows>');
+        } else {
+            $stream = $streams->open($this->path, 'xml', $context->appendSafe(), $partitions);
+        }
+
+        $this->writeXML($rows, $stream);
+    }
+
+    /**
+     * @throws RuntimeException
+     */
+    private function writeXML(Rows $rows, FileStream $stream) : void
+    {
+        if (!\count($rows)) {
+            return;
+        }
+
+        $xml = Writer::inMemory()
+            ->apply(
+                static function (\XMLWriter $xmlWriter) use ($rows) : void {
+                    $data = [];
+
+                    foreach ($rows->toArray() as $row) {
+                        $children = [];
+
+                        foreach ($row as $name => $value) {
+                            $children[] = element($name, value((string) $value));
```

Comment:
`value()` method from Vee requires getting a string, but XML allows integers and other values.

Reply:
wait, how does xml allow for integers? `<node>1</node>` and `<node>test</node>` are always string, even that first one is numeric, it's the schema that defines allowed values, not the file itself, no?